images/dvcr-artifact/pkg/registry/registry.go

```diff
@@ -105,6 +106,95 @@ func NewDataProcessor(ds datasource.DataSourceInterface, dest DestinationRegistr
 	}, nil
 }
 
+func createImgFromDevice(deviceName, imgName string) error {
+	blockDevice, err := os.Open(deviceName)
+	if err != nil {
+		fmt.Errorf("can not get open device")
+		return err
+	}
+	defer blockDevice.Close()
+
+	imgFile, err := os.Create(imgName)
+	if err != nil {
+		fmt.Errorf("can not get create image")
+		return err
+	}
+	defer imgFile.Close()
+
+	buf := make([]byte, 1024*1024)
+	_, err = io.CopyBuffer(imgFile, blockDevice, buf)
+	if err != nil {
+		fmt.Errorf("can not get copy to image")
+		return err
+	}
+
+	return nil
+}
+
+func (p DataProcessor) ProcessFromBlockDevice(ctx context.Context) (ImportRes, error) {
```

Comment:
No need to create additional method here, in DataProcessor, you need to create a new DataSource implementation in pkg/datasource directory. For example, BlockDeviceDataSource. DataProcessor is agnostic to source details, it only requires io.Reader, length and name.

Reply:
Good idea. Thanks.

Fixed.